plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -92,6 +92,7 @@
     private int maxOpenSortFiles = 50;
     private int writeValidationThreads = 16;
     private boolean validateBucketing = true;
+    private boolean usePartitionedBucketFunction = true;
```

Comment:
Is it safe to turn on by default? 
Is it about paralellism only?, or can it create different file sizes as well?

Reply:
It should not change file size, it should only help with parallelism during writes to partitioned and bucketed tables. It's possible that we don't have enough writers to take advantage of the extra distribution, in that case we would have added some extra overhead from hashing by the partition columns.
Based on what we know so far, it seems safe to turn on by default. Having a kill switch should help to mitigate any unforeseen problems.